packages/@livestore/common/src/leader-thread/LeaderSyncProcessor.ts

```diff
@@ -336,6 +344,9 @@ export const makeLeaderSyncProcessor = ({
 
       const backendPushingFiberHandle = yield* FiberHandle.make<void, never>()
       const backendPushingEffect = backgroundBackendPushing({
+        pullLatch,
+        localPushesLatch,
+        syncStateSref,
```

Comment:
I'm reusing local pushes latch here, I think should be ok, perhaps should rename it to just `pushesLatch`. Awaiting on both latches inside.

By the way, I find it a bit weird that pulling locks pushing and pushing locks pulling. Perhaps introduce a third latch like `sync` so that only one process can use that at a time. But there's also ordering to think about like `1) pull 2) local push 2) remote push`. Perhaps introducing a separate class to manage that would be cleaner. Not a high priority though